### PR TITLE
Fix to (re)enable lower-privilege connection strings

### LINKED ISSUE #233: EnsureDatabaseExistsAsync breaks in environments where "master" access disallowed
In companies with stronger security policies, like mine, it may not be possible to give the application login access to the master database. We can run the scripts to manually create the schema/tables but EnsureDatabaseExistsAsync will fail to log into the master database preventing the listeners from starting up:

```[Error] The listener for function 'MyFunction' was unable to start.
Microsoft.Azure.WebJobs.Host.Listeners.FunctionListenerException : The listener for function 'MyFunction' was unable to start. ---> Microsoft.Data.SqlClient.SqlException : Login failed for user '<token-identified principal>'.
   at Microsoft.Data.ProviderBase.DbConnectionPool.CheckPoolBlockingPeriod(Exception e)
   at Microsoft.Data.ProviderBase.DbConnectionPool.CreateObject(DbConnection owningObject,DbConnectionOptions userOptions,DbConnectionInternal oldConnection)
   at Microsoft.Data.ProviderBase.DbConnectionPool.UserCreateRequest(DbConnection owningObject,DbConnectionOptions userOptions,DbConnectionInternal oldConnection)
   at Microsoft.Data.ProviderBase.DbConnectionPool.TryGetConnection(DbConnection owningObject,UInt32 waitForMultipleObjectsTimeout,Boolean allowCreate,Boolean onlyOneCheckConnection,DbConnectionOptions userOptions,DbConnectionInternal& connection)
   at Microsoft.Data.ProviderBase.DbConnectionPool.WaitForPendingOpen()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async DurableTask.SqlServer.SqlDbManager.EnsureDatabaseExistsAsync() at /_/src/DurableTask.SqlServer/SqlDbManager.cs : 183
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async DurableTask.SqlServer.SqlDbManager.AcquireDatabaseLockAsync(Boolean createDatabaseIfNotExists) at /_/src/DurableTask.SqlServer/SqlDbManager.cs : 137
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async DurableTask.SqlServer.SqlDbManager.CreateOrUpgradeSchemaAsync(Boolean recreateIfExists) at /_/src/DurableTask.SqlServer/SqlDbManager.cs : 33
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Extensions.DurableTask.DurableTaskExtension.StartTaskHubWorkerIfNotStartedAsync() at D:\a\_work\1\s\src\WebJobs.Extensions.DurableTask\DurableTaskExtension.cs : 1414
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.Azure.WebJobs.Host.Listeners.FunctionListener.StartAsync(??) at D:\a\_work\1\s\src\Microsoft.Azure.WebJobs.Host\Listeners\FunctionListener.cs : 68
   End of inner exception```

### PR DESCRIPTION
**Pull request by cgillum**
Resolves https://github.com/microsoft/durabletask-mssql/issues/233

The fix is to skip the database existence check if the current connection string does not have permission to log into the master database.

### CODE REVIEW

File: src/DurableTask.SqlServer/DTUtils.cs [hunk 1]
```diff
@@ -196,6 +196,7 @@ public static bool HasPayload(HistoryEvent e)
             return historyEvent.EventType switch
             {
                 EventType.ExecutionStarted => ((ExecutionStartedEvent)historyEvent).Version,
+                EventType.SubOrchestrationInstanceCreated => ((SubOrchestrationInstanceCreatedEvent)historyEvent).Version,
```
Comment (cgillum) [inline src/DurableTask.SqlServer/DTUtils.cs:199]: This is an unrelated change that I happened to notice accidentally.

File: src/DurableTask.SqlServer/Scripts/logic.sql [hunk 2]
```diff
@@ -1444,7 +1444,7 @@ BEGIN
     -- Instance IDs can be overwritten only if the orchestration is in a terminal state
     IF @existingStatus NOT IN ('Failed')
     BEGIN
-        DECLARE @msg nvarchar(4000) = FORMATMESSAGE('Cannot rewing instance with ID ''%s'' because it is not in a ''Failed'' state, but in ''%s'' state.', @InstanceID, @existingStatus);
+        DECLARE @msg nvarchar(4000) = FORMATMESSAGE('Cannot rewind instance with ID ''%s'' because it is not in a ''Failed'' state, but in ''%s'' state.', @InstanceID, @existingStatus);
```
Comment (cgillum) [inline src/DurableTask.SqlServer/Scripts/logic.sql:1447]: Another unrelated change that I happened to notice accidentally.

**Review by davidmrdavid — COMMENTED** [review]
This change seems safe, but I want to make sure I understand the implications. Left some questions to gather context, but otherwise almost ready to approve.

**Comment by cgillum** [comment]
After some discussion, it was decided that this fix isn't needed after all.
